File: app/src/main/java/com/friday/ar/store/pager/CardViewPagerAdapter.java
package com.friday.ar.store.pager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.friday.ar.store.ui.AppCardFragment;
import com.google.firebase.firestore.CollectionReference;

import java.util.ArrayList;

public class CardViewPagerAdapter extends FragmentPagerAdapter {
    private ArrayList<CollectionReference> dataList;

    public CardViewPagerAdapter(FragmentManager fm, ArrayList<CollectionReference> data) {
        super(fm);
        this.dataList = data;
    }

    @Override
    public int getCount() {
        return dataList.size();
    }


    @NonNull
    @Override
    public Fragment getItem(int position) {
        AppCardFragment fragment = new AppCardFragment(dataList.get(position));
        return fragment;
    }
}

File: app/src/main/java/com/friday/ar/store/pager/CardViewPagerAdapter.kt
package com.friday.ar.store.pager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.friday.ar.store.ui.AppCardFragment
import com.google.firebase.firestore.CollectionReference
import java.util.*

class CardViewPagerAdapter(fm: FragmentManager, private val dataList: ArrayList<CollectionReference>) : FragmentPagerAdapter(fm) {

    override fun getCount(): Int {
        return dataList.size
    }


    override fun getItem(position: Int): Fragment {
        return AppCardFragment(dataList[position])
    }
}
